system-addon/lib/TopSitesFeed.jsm

```diff
@@ -432,6 +429,9 @@ this.TopSitesFeed = class TopSitesFeed {
       case at.PREVIEW_REQUEST:
         this.getScreenshotPreview(action.data.url, action.meta.fromTarget);
         break;
+      case at.SCREENSHOT_NEEDED:
+        this._fetchScreenshot(action.data.link, action.data.url);
```

Comment:
> The link object being passed around is from the cache, and it has functions of it via __sharedCache. I guess it might happen to work in Main as the same object is used?? 

Oh, hmm. I see. This isn't working because the `__sharedCache` seems to be getting dropped somewhere. `TypeError: cache is undefined[Learn More] Screenshots.jsm:125:1 `